tests/integration/targets/agent_job_category/tasks/main.yml

```diff
@@ -18,26 +18,28 @@
       register: result
     - assert:
         that:
-          - result.data.ComputerName != None
-          - result.data.InstanceName != None
-          - result.data.SqlInstance != None
-          - result.data.Name == "{{ category_name }}"
-          - result.data.ID != None
-          - result.data.CategoryType == "LocalJob"
-          - result.data.JobCount == 0
+          - result.data != None
+          - (result.data | from_json | json_query('ComputerName')) != None
+          - (result.data | from_json | json_query('InstanceName')) != None
+          - (result.data | from_json | json_query('SqlInstance')) != None
+          - (result.data | from_json | json_query('Name')) == "{{ category_name }}"
+          - (result.data | from_json | json_query('ID')) != None
+          - (result.data | from_json | json_query('CategoryType')) == "LocalJob"
+          - (result.data | from_json | json_query('JobCount')) == 0
```

Comment:
having to use `from_json` is the clue that we aren't returning proper objects this way; the result from a module should be objects (I also think we should avoid `json_query` here as it's not needed, but that's a little more opinionated).